lib/StackAnalysis/StackAnalysis.cpp

```diff
@@ -53,6 +122,272 @@ static opt<std::string> ABIAnalysisOutputPath("abi-analysis-output",
                                               value_desc("path"),
                                               cat(MainCategory));
 
+static opt<std::string> IndirectBranchInfoSummaryPath("indirect-branch-info-"
+                                                      "summary",
+                                                      desc("Write the results "
+                                                           "of SA2 on disk."),
+                                                      value_desc("filename"));
+
+static opt<std::string> AAWriterPath("aa-writer",
+                                     desc("Dump to disk the outlined functions "
+                                          "with annotated alias info."),
+                                     value_desc("filename"));
+
+/// \brief Candidate Function Entry Points structure.
+struct CFEP {
+  CFEP(llvm::BasicBlock *Entry, bool Force) : Entry(Entry), Force(Force) {}
+
+  llvm::BasicBlock *Entry;
+  bool Force;
+};
+
+/// \brief A summary of the analysis of a function.
+///
+///        For each function detected, the following information are included:
+///        its type ("regular", "noreturn" or "fake"), which ABI registers are
+///        overwritten, its control-flow graph, and an elected stack offset (to
+///        tell if the stack pointer is restored at its original position).
+struct FunctionSummary {
+public:
+  model::FunctionType::Values Type;
+  std::set<llvm::GlobalVariable *> ClobberedRegisters;
+  ABIAnalyses::ABIAnalysesResults ABIResults;
+  SortedVector<model::BasicBlock> CFG;
+  std::optional<int64_t> ElectedFSO;
+  llvm::Function *FakeFunction;
+
+public:
+  FunctionSummary(model::FunctionType::Values Type,
+                  std::set<llvm::GlobalVariable *> ClobberedRegisters,
+                  ABIAnalyses::ABIAnalysesResults ABIResults,
+                  SortedVector<model::BasicBlock> CFG,
+                  std::optional<int64_t> ElectedFSO,
+                  llvm::Function *FakeFunction) :
+    Type(Type),
+    ClobberedRegisters(std::move(ClobberedRegisters)),
+    ABIResults(std::move(ABIResults)),
+    CFG(std::move(CFG)),
+    ElectedFSO(ElectedFSO),
+    FakeFunction(FakeFunction) {
+    if (FakeFunction != nullptr)
+      revng_assert(CFG.empty());
+  }
```

Comment:
The way you're using `std::move` is not very appropriate. You're moving an argument passed by value.

Simpler, equivalent solution:

```suggestion
  FunctionSummary(model::FunctionType::Values Type,
                  const std::set<llvm::GlobalVariable *> &ClobberedRegisters,
                  const ABIAnalyses::ABIAnalysesResults &ABIResults,
                  const SortedVector<model::BasicBlock> &CFG,
                  std::optional<int64_t> ElectedFSO,
                  llvm::Function *FakeFunction) :
    Type(Type),
    ClobberedRegisters(ClobberedRegisters),
    ABIResults(ABIResults),
    CFG(CFG),
    ElectedFSO(ElectedFSO),
    FakeFunction(FakeFunction) {
    if (FakeFunction != nullptr)
      revng_assert(CFG.empty());
  }
```

What you were probably trying to achieve here was having a rvalue reference (`&&`) as an argument.
Then you need to `std::move` on the call site.

```suggestion
  FunctionSummary(model::FunctionType::Values Type,
                  std::set<llvm::GlobalVariable *> &&ClobberedRegisters,
                  ABIAnalyses::ABIAnalysesResults &&ABIResults,
                  SortedVector<model::BasicBlock> &&CFG,
                  std::optional<int64_t> ElectedFSO,
                  llvm::Function *FakeFunction) :
    Type(Type),
    ClobberedRegisters(std::move(ClobberedRegisters)),
    ABIResults(std::move(ABIResults)),
    CFG(std::move(CFG)),
    ElectedFSO(ElectedFSO),
    FakeFunction(FakeFunction) {
    if (FakeFunction != nullptr)
      revng_assert(CFG.empty());
  }
```

And then use:
```c++
FunctionSummary(Type, std::move(ClobberedRegs), std::move(ABIResults), std::move(CFG), WinFSO, nullptr);
```